test/compress/issue-973.js

```diff
@@ -44,26 +54,43 @@ this_binding_collapse_vars: {
     input: {
         var c = a; c();
         var d = a.b; d();
+        var e = eval; e();
     }
     expect: {
         a();
         (0, a.b)();
+        (0, eval)();
     }
 }
 
-eval_direct_calls: {
+this_binding_side_effects: {
     options = {
-        side_effects: true,
-        collapse_vars: true
-    }
+        side_effects : true
+    };
     input: {
-        (0, eval)('');
-
-        var fn = eval;
-        fn('');
+        (function (foo) {
+            (0, foo)();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
+        (function (foo) {
+            var eval = console;
+            (0, foo)();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
     }
     expect: {
-        (0, eval)('');
-        (0, eval)('');
+        (function (foo) {
+            foo();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
+        (function (foo) {
+            var eval = console;
+            foo();
+            (0, foo.bar)();
+            eval('console.log(foo);');
```

Comment:
Optimizing `(0, x)()` into `x()` is completely safe, *unless* `x` is `eval`. I.e.:

```
var x = eval;
// these calls are equal
(0, x)();
x();

var eval = global.eval;
// these calls are *not* equal
(0, eval)();
eval();